### PR TITLE
Adds ChecksumType element and relative Item Type w Enums, updates wha…

### PR DESCRIPTION
**Pull request by kevinkirkl3y**
…tsnew

### CODE REVIEW

**Review by cwhanse — APPROVED** [review]

**Review by jrippingale — CHANGES_REQUESTED** [review]
Kevin, please try to add descriptions of the source of the different checksum types.  Thank you!

**Comment by kevinkirkl3y** [comment]
> Kevin, please try to add descriptions of the source of the different checksum types. Thank you!

@jrippingale @cwhanse  I found the references here: https://csrc.nist.gov/glossary?sortBy-lg=relevance&ipp-lg=100

**Review by jrippingale — APPROVED** [review]
Lovely descriptions!

File: Master-OB-OpenAPI.json [hunk 1]
```diff
@@ -18894,6 +18914,35 @@
           "description": ""
         }
       }
+    },
+    "ChecksumTypeItemType": {
+      "description": "",
+      "enums": {
+        "CRC": {
+          "label": "CRC",
+          "description": "Cyclical Redundancy Check - NIST SP 800-72"
```
Comment (cwhanse) [inline Master-OB-OpenAPI.json:18923]: This defines the term Cyclical Redundancy Check but there are literally dozens of such checks. Knowing it's a CRC is probably not enough information to decode a checksum.

File: Master-OB-OpenAPI.json [hunk 2]
```diff
@@ -18894,6 +18914,35 @@
...
+        },
+        "MD5": {
+          "label": "MD5",
+          "description": "Message Digest 5 - NIST SP 800-45 Version 2, NIST SP 800-101 Rev. 1"
```
Comment (cwhanse) [inline Master-OB-OpenAPI.json:18927]: I think we should reference RFC 1321, April 1992.  Those NIST documents are guidelines on email and device security.

File: Master-OB-OpenAPI.json [hunk 3]
```diff
@@ -18894,6 +18914,35 @@
...
+        },
+        "SHA1": {
+          "label": "SHA1",
+          "description": "Secure Hash Algorithm 1 - NIST SP 800-22 Rev. 1a"
```
Comment (cwhanse) [inline Master-OB-OpenAPI.json:18931]: Reference "FIPS 180-2"

File: Master-OB-OpenAPI.json [hunk 4]
```diff
@@ -18894,6 +18914,35 @@
...
+        },
+        "SHA2": {
+          "label": "SHA2",
+          "description": "Secure Hash Algorithm 2 - CNSSI 4009-2015"
```
Comment (cwhanse) [inline Master-OB-OpenAPI.json:18935]: Reference "FIPS 180-2"

File: Master-OB-OpenAPI.json [hunk 5]
```diff
@@ -18894,6 +18914,35 @@
...
+        },
+        "SHA3": {
+          "label": "SHA3",
+          "description": "Secure Hash Algorithm 3 - NIST SP 800-185"
```
Comment (cwhanse) [inline Master-OB-OpenAPI.json:18939]: Reference "FIPS 180-2"

File: Master-OB-OpenAPI.json [hunk 6]
```diff
@@ -18894,6 +18914,35 @@
...
+        },
+        "Keccak_256": {
+          "label": "Keccak-256",
+          "description": "The family of all sponge functions with a KECCAK-f permutation as the underlying function and multi-rate padding as the padding rule. KECCAK was originally specified in [2], and standardized in FIPS 202. -  NIST SP 800-185"
```
Comment (cwhanse) [inline Master-OB-OpenAPI.json:18943]: Reference "FIPS 202" I wouldn't try to say more.

**Review by cwhanse — APPROVED** [review]

**Pull request merged**